packages/iterator-utilities/src/lib/windows.ts

```diff
@@ -0,0 +1,39 @@
+import { from, type IterableResolvable } from './from';
+import { assertPositive } from './shared/_assertPositive';
+import { makeIterableIterator } from './shared/_makeIterableIterator';
+import { toIntegerOrInfinityOrThrow } from './shared/_toIntegerOrInfinityOrThrow';
+
+/**
+ * Creates an iterable with arrays of `count` elements representing a sliding window.
+ *
+ * @param iterable The iterator to take values from.
+ * @param count The maximum number of values in the window.
+ * @returns An iterator that yields windows with `count` values from the provided iterator.
+ *
+ * @example
+ * ```typescript
+ * import { windows } from '@sapphire/iterator-utilities';
+ *
+ * const iterable = [1, 2, 3, 4, 5];
+ * console.log([...windows(iterable, 2)]);
+ * // Output: [[1, 2], [2, 3], [3, 4], [4, 5]]
+ * ```
+ */
+export function windows<const ElementType>(iterable: IterableResolvable<ElementType>, count: number): IterableIterator<ElementType[]> {
```

Comment:
Funny pun but shouldn't this be called `window` not `windows`?

Reply:
No, it's a method that makes windows, not a single window.

Also the name is based on the proposal: https://github.com/tc39/proposal-iterator-chunking